nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -275,12 +276,12 @@ def reset_tools(self):
         else:
             self.data_collector.reset()
         if self.metrics_calculator is None:
-            self.metrics_calculator = NormMetricsCalculator(p=self.p, dim=0)
+            self.metrics_calculator = NormMetricsCalculator(p=self.p, scalors=Scaling(kernel_size=[1], kernel_padding_mode='back'))
         if self.sparsity_allocator is None:
```

Comment:
what is the relation between `metrics_calculator` and `sparsity_allocator`? I thought `metrics_calculator` is a member of sparsity_allocator.